### PR TITLE
Extract routers logic into controllers

### PR DESCRIPTION
**Pull request by sin6pi7**

### CODE REVIEW

File: packages/backend/src/api/routers/FrontendRouter.ts [hunk 1]
```diff
@@ -1,95 +1,33 @@
...
-        timestamp: update.timestamp,
-        positionCount: update.positionCount,
-      })),
-      fullCount: Number(fullCount),
-      params: {
-        page,
-        perPage,
-      },
-    })
+    if ([page, perPage].some(Number.isNaN)) {
```
Comment (sz-piotr) [inline packages/backend/src/api/routers/FrontendRouter.ts:16]: We'd also want to change this logic so that we parse `ctx.query` using zod. This can be implemented as part of another PR.

**Review by sz-piotr — CHANGES_REQUESTED** [review]

File: packages/backend/src/api/routers/FrontendRouter.ts (hunk 1 above)
Comment (sin6pi7) [reply packages/backend/src/api/routers/FrontendRouter.ts:16]: These were my thoughts as well. However, the router which is responsible for request data validation and parsing was not doing that for this specific route (unlike other routes where `PedersenHash` and `BigInt` constructors were handling it), so I added it to stay consistent. Otherwise we rely on the controller.
